Update devconsole empty state pages

rottencandy opened the pull request:
**Fixes**: 
https://issues.redhat.com/browse/ODC-5857
<!-- For e.g Fixes: https://issues.redhat.com/browse/ODC-XXX -->

**Analysis / Root cause**: 
Some pages in devconsole were not consistent for when the user does not have access to any project.
<!-- Briefly describe analysis of US/Task or root cause of Defect -->

**Solution Description**: 
- Show "getting started" notification when user with no project access navigates to topology. Do not show Add page.
- Show projects list in search page when user is not in a specific namespace.
- Show getting started notification on Add page for user with no project access.
<!-- Describe your code changes in detail and explain the solution -->

**Screen shots / Gifs for design review**: 
Topology - no accessible projects:
![1](https://user-images.githubusercontent.com/20013884/125065580-9950e600-e0cf-11eb-9120-1e88eff9749c.png)
Search - project not selected:
![1](https://user-images.githubusercontent.com/20013884/125096478-d084be80-e0f2-11eb-8b55-8256d8e5a2b0.png)
+Add - no accessible projects:
![0](https://user-images.githubusercontent.com/20013884/125065637-b08fd380-e0cf-11eb-90d5-55f4ba0d5025.png)

@openshift/team-devconsole-ux 
<!-- If change affects UI in any way, tag @openshift/team-devconsole-ux and add screenshots/gifs  -->

**Browser conformance**: 
<!-- To mark tested browsers, use [x] -->
- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

/kind bug

rottencandy commented:
/cc @bgliwa01

andrewballantyne reviewed (changes requested):
Got TS errors around the `withStartGuide`

Also, your Search page is missing out on the project selector/secondary masthead:
![image](https://user-images.githubusercontent.com/8126518/123703527-7e11eb00-d832-11eb-8ef5-6307c03492b2.png)

rottencandy commented:
> Got TS errors around the `withStartGuide`
> 
> Also, your Search page is missing out on the project selector/secondary masthead

Fixed Type errors, and added namespace selector:
![2](https://user-images.githubusercontent.com/20013884/123772615-9db81a80-d8e9-11eb-9321-5cb3e8419809.png)

rohitkrai03 commented:
@rottencandy Seems like this UX doc suggests we should reuse current topology empty state with the getting started section on top. But that will have an issue because we have options like topology quick add and add to project links in topology empty states enabled. In this case when there are no projects, those options cannot be enabled. @bgliwa01 WDYT?

rottencandy commented on frontend/packages/topology/src/components/page/TopologyPage.tsx:
```diff
@@ -105,27 +100,25 @@ const TopologyPage: React.FC<TopologyPageProps> = ({
             viewType === TopologyViewType.graph ? 'topology-graph-page' : 'topology-list-page'
           }
         >
-          <ProjectsExistWrapper title={t('topology~Topology')} projects={projects}>
```
Updated Add page to use `withStartGuide` and deleted `ProjectsExistWrapper`.

rottencandy commented:
/assign @rohitkrai03

rohitkrai03 reviewed (approved):
/lgtm

andrewballantyne commented:
/lgtm cancel

bgliwa01 commented:
The secondary masthead with the project selector should be above the blue box. Thanks!

andrewballantyne commented on frontend/packages/dev-console/src/components/add/AddPage.tsx:
```diff
@@ -25,28 +25,24 @@ const AddPage: React.FC<AddPageProps> = ({ match }) => {
         <title data-test-id="page-title">{`+${t('devconsole~Add')}`}</title>
       </Helmet>
       <NamespacedPage>
```
We need this component to be not inside the `withStartGuide` HOC.

andrewballantyne commented on frontend/packages/topology/src/components/page/TopologyPage.tsx:
```diff
@@ -34,7 +32,7 @@ interface TopologyPageProps {
 }
 
-const TopologyPage: React.FC<TopologyPageProps> = ({
+export const TopologyPage: React.FC<TopologyPageProps> = ({
```
Same problem for this guy, but I can't put a comment on line 90 🙂

rottencandy replied:
Updated. Moved blue box to below project dropdown in both places.

andrewballantyne commented on frontend/packages/dev-console/src/components/SearchPage.tsx:
```diff
@@ -0,0 +1,40 @@
+  );
+};
+
+export default Search;
```
Did we not want the `withStartGuide` here?

rottencandy replied:
Updated.
![1](https://user-images.githubusercontent.com/20013884/125096862-29545700-e0f3-11eb-95ae-62d64402f5bd.png)

andrewballantyne reviewed (changes requested):
Only question I have is do we want to setup Search like the other pages?

andrewballantyne reviewed (approved):
/lgtm
